rewriter/rewriter.cc

```diff
@@ -129,6 +130,8 @@ class Rewriterer {
                     }
                 },
 
+                [&](ast::MethodDef *mdef) { Initializer::run(ctx, mdef, prevStat); },
```

Comment:
Note that this `typecase` only operates on method defs at the top-level of the class.

I'm not sure at what point methods are flattened anymore (I think after?) but this means that your new pass won't run on nested methods.

That's probably fine, I think it's a pathological case, but I think there's also a quick fix: you should be able to move the `Initializer::run` pass to a `postTransformMethodDef`, sibling to the `postTransformClassDef` and `postTransformSend` that already exist in this file.

Reply:
Oh, that's a good point. This _does_ run pre-flattening, so I'll do that while simultaneously hoping that we never encounter code where this wouldn't work.

Reply:
Actually, I take that back: that fix is not as quick as I'd hope. The problem is that this needs to know about the previous expression as well, because that's the `sig`. We already implement logic to handle that in this specific `typecase` but we don't in general, and if I just naively moved `Initializer::run` to `postTransformMethodDef` then I'd need to implement a _lot_ more logic in order to keep track of "what was the previous expression so that I can make sure it's a `sig`?"

As it is, the pathological cases are going to look something like
```ruby
# typed: true
class A
  extend T::Sig

  sig {void}
  def self.foo
    sig {params(x: Integer).void}
    def initialize(x)
      @x = x
    end
  end
end
```
and I think they're unlikely enough that my instinct now is to leave it as-is and only do the extra work if we _really_ need to.

Reply:
Can you add this snippet as a test case?

Reply:
Added to the `initializer` test file.